lib/rexml/element.rb

```diff
@@ -969,6 +995,30 @@ def [](name_or_index)
       end
     end
 
+
+    # :call-seq:
+    #   attribute(name, namespace = nil)
+    #
+    # Returns the string value for the given attribute name.
+    #
+    # With only argument +name+ given,
+    # returns the value of the named attribute if it exists, otherwise +nil+:
+    #
+    #   xml_string = <<-EOT
+    #     <root xmlns="ns0">
+    #       <a xmlns="ns1" attr="value"></a>
+    #       <b xmlns="ns2" attr="value"></b>
+    #       <c attr="value"/>
+    #    </root>
+    #   EOT
+    #   d = REXML::Document.new(xml_string)
+    #   root = d.root
+    #   a = root[1] # => <a xmlns='ns1' attr='value'/>
+    #   a.attribute('attr') # => attr='value'
+    #   a.attribute('nope') # => nil
+    #
+    # TODO:  Make sensible examples using both arguments.
```

Comment:
How about this?

```text
require "rexml/document"

xml = <<-XML
<root xmlns:a='a' a:x='a:x' x='x'/>
XML
document = REXML::Document.new(xml)
document.root.attribute("x")      # => x='x'
document.root.attribute("x", "a") # => a:x='a:x'
```

Reply:
Added.  Please see whether you think the accompanying text is adequate.

Reply:
Confirmed. It looks good to me.